gateway/src/api/latest.rs

```diff
@@ -361,11 +362,23 @@ async fn delete_project(
         let resources: Vec<shuttle_common::resource::Response> =
             serde_json::from_slice(&body_bytes)
                 .map_err(|e| Error::source(ErrorKind::Internal, e))?;
-        if resources
+
+        let resources = resources
             .into_iter()
-            .any(|s| matches!(s.r#type, shuttle_common::resource::Type::Database(_)))
-        {
-            return Err(Error::from_kind(ErrorKind::ProjectHasDatabase));
+            .filter(|resource| {
+                matches!(
+                    resource.r#type,
+                    shuttle_common::resource::Type::Database(_)
+                        | shuttle_common::resource::Type::Secrets
+                )
+            })
+            .map(|resource| resource.r#type.to_string())
+            .collect::<HashSet<_>>()
```

Comment:
This may be a bit overkill (it needs to add a few `Derive(Hash)` annotations), there shouldn't be more than one of each resource, but we have seen that it can happen.

Reply:
In that case, it might be better to not dedup them here, so that the issue is not obfuscated.

Reply:
> This may be a bit overkill (it needs to add a few `Derive(Hash)` annotations), there shouldn't be more than one of each resource, but we have seen that it can happen.

Not sure why it matters if there is more than on resource attached for checking if the project has resources.

Reply:
I'll drop it, thanks!